File: Python/EntradaCommandLine/6_SumN/Sequential/sum_n.py
#!/usr/bin/env python3
"""
SumN sequential - EntradaCommandLine (Python)
Usage: python sum_n.py <n>
"""
import sys
import time

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print(f"Usage: {sys.argv[0]} <n>")
        sys.exit(1)
    n = int(sys.argv[1])
    print('--------- SumN&Sequential (CommandLine - Python) --------')
    start = time.process_time()
    s = sum(range(1, n+1))
    end = time.process_time()
    print(f'Sum of first {n} naturals = {s}')
    print(f'Tiempo total de ejecución (CPU): {end - start:.6f} segundos')

File: C/EntradaCommandLine/6_SumN/Sequential/SumN.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/* -----------------------------------------------------------------------------
 * Programa: SumN&Sequential
 * -------------------------
 * Este programa calcula la suma de los primeros n números naturales de manera
 * secuencial y muestra los números junto con su suma total. También mide el
 * tiempo de ejecución de la operación.
 * -----------------------------------------------------------------------------
 */

int main(int argc, char **argv) {
    if (argc < 2) {
        fprintf(stderr, "Usage: %s <n>\n", argv[0]);
        return 1;
    }
    int n = atoi(argv[1]); // Cantidad de números naturales a sumar
    int sum = 0; // Variable acumuladora para la suma

    printf("--------- SumN&Sequential (CommandLine) --------\n");

    // Iniciar medición de tiempo
    clock_t start = clock();

    // Mostrar los números y calcular la suma
    printf("\nLos primeros %d numeros naturales son:\n", n);
    for (int i = 1; i <= n; i++) {
        sum += i;      // Acumular suma
        printf("%d ", i); // Imprimir número actual
    }

    // Finalizar medición de tiempo
    clock_t end = clock();

    // Mostrar el resultado final y tiempo de ejecución
    printf("\n\nLa suma de los %d primeros numeros naturales es: %d\n", n, sum);
    printf("Tiempo total de ejecución: %.6f segundos\n",
           (double)(end - start) / CLOCKS_PER_SEC);

    printf("---------- Program End ----------\n");
    return 0;
}
